File: sdks/python-ws/examples/streaming.py
"""
Streaming example: Generate text with streaming tokens
"""

import asyncio
from realm import RealmWebSocketClient


async def main():
    client = RealmWebSocketClient(url="ws://localhost:8080")
    
    try:
        await client.connect()
        print("✓ Connected!")

        print("\nGenerating with streaming...")
        
        # Note: Full streaming implementation pending server support
        # For now, this will yield the complete result
        async for token in client.generate_stream({
            "prompt": "Tell me a short story about a robot.",
            "max_tokens": 100,
            "temperature": 0.8,
        }):
            print(token, end="", flush=True)
        
        print("\n✓ Complete!")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        await client.disconnect()


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/nodejs-ws/src/examples/streaming.ts
/**
 * Streaming example: Generate text with streaming tokens
 */

import { RealmWebSocketClient } from "../client";

async function main() {
  const client = new RealmWebSocketClient({
    url: "ws://localhost:8080",
    model: "tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf",
    apiKey: process.env.REALM_API_KEY,
  });

  try {
    await client.connect();
    console.log("✓ Connected!");

    console.log("\nGenerating with streaming...");
    
    for await (const token of client.generateStream({
      prompt: "Tell me a short story about a robot.",
      max_tokens: 100,
      temperature: 0.8,
    })) {
      process.stdout.write(token);
    }
    
    console.log("\n✓ Complete!");

  } catch (error) {
    console.error("Error:", error);
  } finally {
    client.disconnect();
  }
}

main().catch(console.error);

